### DISCUSSION TITLE
Using Alchemy with the Injected Connector configuration

### DISCUSSION THREAD

**Discussion by connorphill** [wevm/wagmi#338] [Q&A] [answered]
I am fairly new to using wagmi and have been trying to understand the different parts of the injected wallet configuration, but can't figure out how to use my Alchemy API connection and a few parts about the wallet connection setup.

For starters, I logged the `defaultChains` to see what chains are accessible and I noticed the rpcUrls for each test/mainnet use infura. Since I am using an Alchemy account should I overwrite this to use the alchemy URLs or is that not necessary as everyone can interact the EVMs hosted on infura without an API key?

```
[
  {
    id: 1,
    name: 'Mainnet',
    nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
    rpcUrls: [ 'https://mainnet.infura.io/v3' ],
    blockExplorers: [ [Object] ]
  },
  {
    id: 3,
    name: 'Ropsten',
    nativeCurrency: { name: 'Ropsten Ether', symbol: 'ropETH', decimals: 18 },
    rpcUrls: [ 'https://ropsten.infura.io/v3' ],
    blockExplorers: [ [Object] ],
    testnet: true
  },
  ...
]
```

I noticed in the documentation, https://wagmi.sh/docs/provider#provider-optional, that there is an overwrite behavior for provider, is this where I should be applying the Alchemy information? Any location in github repo that points to a `.AlchemyProvider()`?

`const provider = ({ chainId }) => new providers.InfuraProvider(chainId, 'Your infura id')`

**Comment by swene-dev** [comment]
I don't know but this code from the examples shows how to setup 3 providers:
[NextJS example](https://github.com/tmm/wagmi/blob/main/examples/next/src/pages/_app.tsx)

**Comment by connorphill** [comment, marked as answer]
Something that I missed in the documentation is the direct reference of the ethers package and in that packages code there is an [`AlchemyProvider()` method](https://docs.ethers.io/v5/api/providers/api-providers/#AlchemyProvider). This resolved my issue.

[wagmi Documentation - provider (optional)](https://wagmi.sh/docs/provider#provider-optional)

My updated code:

```
import { providers } from 'ethers'
import { Provider, defaultChains } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'

const provider = ({ chainId }) => new providers.AlchemyProvider(chainId, process.env.ALCHEMY_API);
```
